core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2380,6 +2380,18 @@ class WorkspaceService(protected val userInfo: UserInfo,
     } yield result
   }
 
+  def getBillingProfileId(billingProjectName: String): Future[Option[String]] = {
```

Comment:
I thought of using `withBillingProjectContext`, which is at line 2367. However, that doesn't work on Azure billing projects because of `failUnlessBillingAccountHasAccess(billingProject, parentSpan)`.

I do wonder though if we want to thread a span through all of this for logging.

Reply:
Agreed about adding a span. Maybe we make the `failUnlessBillingAccountHasAccess` resilient to non-gcp projects? I'm fine scoping that out for this effort though.

Reply:
We could potentially change `withBillingProjectContext` to only call `failUnlessBillingAccountHasAccess` if `billingProfileId` is not defined.